### ISSUE TITLE
An empty whitelist produces different results than no whitelist for HillClimbSearch

### ISSUE THREAD

**Issue by kyle-pena-nlp** [pgmpy/pgmpy#1284]
@ankurankan --- Posting as a new issue since the other issue got closed.  I pulled dev/latest and ran some tests.  Something is off with the whitelist functionality in the HillClimbSearch for Bayesian Networks.  I have a fix locally, happy to discuss first and then submit a PR.  I'm also fairly certain that this same bug makes it possible for whitelisted edges to get removed by a flip operation.

## Description
For a HillClimbSearch, specifying an empty whitelist does not reuslt in the same behavior as specifying no whitelist.  More generally, I *think* what's going on is that specifying an empty whitelist will cause HillClimbSearch to never do any flip operations.

I found this issue while attempting to show that the whitelist logic for the flip operation was still broken in dev.  Since this bug originates with the same line of code, I figured it's sufficient to post this test.

I'd be happy to submit a PR, but I'd like to discuss how the _legal_operations algorithm guarantees edges in the whitelist will be included (esp. if max_iters < len(whitelist) ) 

pgmpy 0.1.10dev
Python 3.8.0
Windows 10

## Expected Behavior
The networks should be identical whether you pass an empty whitelist or no whitelist at all.

## Actual Behavior
The networks are not identical

```
import pgmpy
import pandas as pd
from pgmpy.estimators.HillClimbSearch import HillClimbSearch
from pgmpy.estimators import K2Score
from pgmpy.models import BayesianModel
import numpy as np

X = np.asarray(["A", "B", "A", "B", "A",
    "B", "C", "C", "B", "A",
    "C", "B", "A", "C", "B",
    "D", "E", "D", "E", "D",
    "A", "B", "C", "D", "E"]).reshape((5,-1))

# synthesize data
X = pd.DataFrame({"A": X[:,0], "B": X[:,1], "C": X[:,2], "D": X[:,3], "E": X[:,4]})


# Initialize the starting model and hill climb search
model = BayesianModel([("A", "B"), ("C", "A"), ("E", "A"), ("D", "B"), ("D", "C"), ("D", "E")])
model.add_nodes_from(list(X.columns.values))
search = HillClimbSearch(data = X, scoring_method=K2Score(X))

# Kick off the hill climb search
m1 = search.estimate(start = model, max_indegree = 2, max_iter = 100, white_list = [])
print((m1.edges()))
m2 = search.estimate(start = model, max_indegree = 2, max_iter = 100)
print((m2.edges()))

assert set(m1.edges()) == set(m2.edges()) # Fails
```

**Comment by ankurankan** [comment]
@kyle-pena-nlp Thanks for reporting this. This indeed looks like a bug, would be great if you could open a PR. 

The `white_list` argument is supported to give a list of edges to which the search is limited to, so we don't need to guarantee that they are included in the final model. But seems like there is a bug in that as well, as passing an empty list as an argument for `white_list` still gives a model with edges.

**Comment by ankurankan** [comment]
After #1312
```
In [3]: m1.edges()     
Out[3]: OutEdgeView([('A', 'B'), ('C', 'A'), ('C', 'D'), ('E', 'A'), ('D', 'E')])

In [4]: m2.edges()     
Out[4]: OutEdgeView([('A', 'B'), ('C', 'A'), ('C', 'D'), ('E', 'A'), ('D', 'E')])
```
